### PR TITLE
Screenshots from plugin event on(after:screenshot)

### PR DESCRIPTION
**Pull request by thomaswinkler**
Missing screenshots in Report Portal has been probably the most critical issue I've ran into after #135. Depending on configuration and the structure of the test, no screenshots would be uploaded at all. 

Checking implementation and [Cypress Screenshot Naming Conventions](https://docs.cypress.io/api/commands/screenshot#Naming-conventions), I first tried to implement missing naming conventions extending the glob pattern based approach using regexp. This helped improving results, but still it was not possible to match all possible screenshots from the test. Could provide more detail, but sure you are aware of missing screenshots as for example from `after each` or sub suites, retry attempts, etc.

I was looking for an approach to send the screenshot when Cypress creates the screenshot, ideally getting the absolute path provided by Cypress without any globbing or matching at the end of the test at all. The approach proposed in the PR is using  `after:screenshot` plugin event. It reduces complexity dramatically, as it basically just passes the screenshotInfo (incl. absolute path) to new `reporter.sendScreenshot(screenshotInfo)` function. I am using this for some time now in my setup and never missed a screenshot. 

Another advantage of this approach is the order in the logs. Screenshots no longer appear at the end of the test, but at the right position of log. It would be also possible to use `takenAt` property from screenshotInfo and pass this to Report Portal. 

Please note, I also added a new config option `logScreenshotInfo` to enable logging of the screenshotInfo passed from plugin event. I really helps documenting the screenshot details in Report Portal. I understand not everyone might want this, but really think this could be valuable and need it myself.

### CODE REVIEW

**Comment by thomaswinkler** [comment]
Added new `config.screenshotLogFn` option to replace `config.logScreenshotInfo` introduced in initial commit to allow full customization of screenshot log message. Previous approach was not flexible enough. 

```typescript
reportportalOptions.screenshotLogFn = (screenshotInfo) => {
  const filename = screenshotInfo.path.split('/').pop() 
  return `cy:screenshot\t${filename}\n${JSON.stringify(screenshotInfo, undefined, 2)}`;
};
registerReportPortalPlugin(on, reportportalOptions);
```

File: lib/worker.js [hunk 1]
```diff
@@ -90,9 +90,13 @@ process.on('message', (message) => {
     case reporterEvents.SET_TEST_CASE_ID:
       reporter.setTestCaseId(message.testCaseIdInfo);
       break;
-    case reporterEvents.CUSTOM_SCREENSHOT:
-      reporter.saveCustomScreenshotFilename(message.screenshotInfo);
+    case reporterEvents.SCREENSHOT: {
+      const screenshotInfo = message.screenshotInfo;
+      const logMessage = screenshotInfo.logMessage;
+      delete screenshotInfo.logMessage;
```
Comment (AmsterGet) [inline lib/worker.js:96]: What is the reason of extracting `logMessage` from `screenshotInfo` if it is not changed?
The `reporter.sendScreenshot` called only here and I suppose that `logMessage` can be used directly from `screenshotInfo`.

**Review by AmsterGet — COMMENTED** [review]
Really like your solution as it greatly reduces the complexity and increases the reliability of attaching screenshots! Thank you!
Ready to merge, just have one question (see below).

File: lib/worker.js (hunk 1 above)
Comment (thomaswinkler) [reply lib/worker.js]: @AmsterGet Just pushed a commit to address the issue. Should be more clear with latest changes.

**Review by AmsterGet — APPROVED** [review]

**Pull request merged**